src/Umbraco.Web.UI.Client/src/views/languages/edit.controller.js

```diff
@@ -20,150 +19,118 @@
 
         function init() {
+            vm.loading = true;
 
-            // localize labels
-            var labelKeys = [
+            var promises = [];
+
+            // Localize labels
+            promises.push(localizationService.localizeMany([
                 "treeHeaders_languages",
-                "languages_mandatoryLanguage",
-                "languages_mandatoryLanguageHelp",
-                "languages_defaultLanguage",
-                "languages_defaultLanguageHelp",
                 "languages_addLanguage",
-                "languages_noFallbackLanguageOption",
-                "languages_fallbackLanguageDescription",
-                "languages_fallbackLanguage",
                 "defaultdialogs_confirmSure"
-            ];
-
-            localizationService.localizeMany(labelKeys).then(function (values) {
+            ]).then(function (values) {
                 vm.labels.languages = values[0];
-                vm.labels.mandatoryLanguage = values[1];
-                vm.labels.mandatoryLanguageHelp = values[2];
-                vm.labels.defaultLanguage = values[3];
-                vm.labels.defaultLanguageHelp = values[4];
-                vm.labels.addLanguage = values[5];
-                vm.labels.noFallbackLanguageOption = values[6];
-                vm.labels.areYouSure = values[9];
-
-                $scope.properties = {
-                    fallbackLanguage: {
-                        alias: "fallbackLanguage",
-                        description: values[7],
-                        label: values[8]
-                    }
-                };
+                vm.labels.addLanguage = values[1];
+                vm.labels.areYouSure = values[2];
 
                 if ($routeParams.create) {
-                    vm.page.name = vm.labels.addLanguage;
+                    vm.isNew = true;
+                    vm.language.name = vm.labels.addLanguage;
                 }
-            });
-
-            vm.loading = true;
-
-            var promises = [];
+            }));
 
-            //load all culture/languages
+            // Load all culture/languages
             promises.push(languageResource.getCultures().then(function (culturesDictionary) {
-                var cultures = [];
-                angular.forEach(culturesDictionary, function (value, key) {
-                    cultures.push({
-                        name: key,
-                        displayName: value
-                    });
-                });
-                vm.availableCultures = cultures;
+                vm.availableCultures = culturesDictionary;
             }));
 
-            //load all possible fallback languages
+            // Load all possible fallback languages
             promises.push(languageResource.getAll().then(function (languages) {
                 vm.availableLanguages = languages.filter(function (l) {
                     return $routeParams.id != l.id;
                 });
-                vm.loading = false;
```

Comment:
This disabled the loading indicator before both cultures and current language were finished loading.